src/Symfony/Component/RateLimiter/Policy/SlidingWindowLimiter.php

```diff
@@ -74,7 +74,14 @@ public function reserve(int $tokens = 1, ?float $maxTime = null): Reservation
             if ($availableTokens >= $tokens) {
                 $window->add($tokens);
 
-                $reservation = new Reservation($now, new RateLimit($this->getAvailableTokens($window->getHitCount()), \DateTimeImmutable::createFromFormat('U', floor($now)), true, $this->limit));
+                if ($availableTokens === $tokens) {
+                    $resetDuration = $window->calculateTimeForTokens($this->limit, $window->getHitCount());
+                    $retryAfter = $now + $resetDuration;
+                } else {
+                    $retryAfter = $now;
+                }
+
+                $reservation = new Reservation($now, new RateLimit($this->getAvailableTokens($window->getHitCount()), \DateTimeImmutable::createFromFormat('U', floor($retryAfter)), true, $this->limit));
```

Comment:
Hi!

I was wondering if we were following the spirit and formalism of [the previous lines of code 69 to 72](https://github.com/symfony/symfony/pull/54163/files#diff-db14f1704ceccbe85452aa82a9ab0db9a7836bdbda66fa80bd24ba718c6539b4R69-R72) :

```php
                $resetDuration = $window->calculateTimeForTokens($this->limit, $window->getHitCount());
                $resetTime = \DateTimeImmutable::createFromFormat('U', $availableTokens ? floor($now) : floor($now + $resetDuration));

                return new Reservation($now, new RateLimit($availableTokens, $resetTime, true, $this->limit));
```

it might be possible to propose the following code:

```suggestion
                $resetDuration = $window->calculateTimeForTokens($this->limit, $window->getHitCount());
                $resetTime = \DateTimeImmutable::createFromFormat('U', $availableTokens === $tokens ? floor($now + $resetDuration) : floor($now));

                $reservation = new Reservation($now, new RateLimit($this->getAvailableTokens($window->getHitCount()), $resetTime, true, $this->limit));
```

Do you think it's worth keeping this consistency or do you think it's too much?
(This suggestion passes the tests.)